pyinteraph/correlation/residue_coordinate_matrix.py

```diff
@@ -0,0 +1,49 @@
+from collections import OrderedDict
+import functools
+
+import numpy as np
+import MDAnalysis as mda
+
+from pyinteraph.core import logger
+
+
+class ResidueCoordinateMatrix:
+    def __init__(self, ref: str, traj: str, atoms: str, backbone: bool) -> None:
+        self.ref = ref
+        self.traj = traj
+        self.selected_atoms = "backbone" if backbone else f"name {atoms.replace(',', ' ')}"
```

Comment:
usually these measures are calculated on just CAs rather than backbone.

Even then, it would be good to have e.g. a default selection option (all CA atoms in protein) or the possibility to give a custom MDAnalysis selection string, akin to what we do in groups for hydrogen bonds